modules/gapi/src/backends/sycl/gsyclbackend.hpp

```diff
@@ -0,0 +1,182 @@
+    // List of all resources in graph (both internal and external)
+    std::vector<ade::NodeHandle> m_dataNodes;
+
+    // Actual data of all resources in graph (both internal and external)
+    Mag m_res;
+    GArg packArg(const GArg& arg);
+    sycl::queue m_queue;
+
+public:
+    GSYCLExecutable(const ade::Graph  &graph,
+                    const std::vector<ade::NodeHandle> &nodes);
+
+    // FIXME: Can this be made reshapable?
+    virtual inline bool canReshape() const override { return false; }
+    inline void reshape(ade::Graph&, const GCompileArgs&) override
+    {
+        util::throw_error(std::logic_error("GSYCLExecutable::reshape() should never be called"));
+    }
+
+    virtual void run(std::vector<InObj> &&input_objs,
+                     std::vector<OutObj> &&ouput_objs) override;
+
+    void initSYCLContext();
+};
+
+template <typename T>
+int syclTypeToMatType()
+    {
+        if (std::is_same<T, unsigned char>::value){
+            return CV_8U;
+        }
+        else if (std::is_same<T, signed char>::value){
+            return CV_8S;
+        }
+        else if (std::is_same<T, unsigned int>::value){
+            return CV_16U;
+        }
+        else if (std::is_same<T, int>::value){
+            return CV_16S;
+        }
+        else if (std::is_same<T, long int>::value){
+            return CV_32S;
+        }
+        else if (std::is_same<T, float>::value){
+            return CV_32F;
+        }
+        else if (std::is_same<T, double>::value){
+            return CV_64F;
+        }
+    };
+
+struct GAPI_EXPORTS SYCLBufferDesc
+    {
+        int depth;
+        int chan;
+        cv::Size size;
+        bool planar;
+    };
+
+template <typename T, int Dimensions, typename AllocatorT, typename Enable>
+SYCLBufferDesc getSYCLBufferDesc(sycl::buffer<T, Dimensions, AllocatorT, Enable>& buffer)
+    {
+        SYCLBufferDesc desc;
+        desc.depth = sizeof(T)*8; // size in bits
+        desc.chan = buffer.get_range().get(2); // assume channel dim is third
+        desc.size = cv::Size(buffer.get_range().get(0), buffer.get_range().get(1));
+        if (desc.chan == 1)
+        {
+            desc.planar = true;
+        }
+        else
+        {
+            desc.planar = false;
+        }
+        return desc;
+    };
+
+cv::GMatDesc getGMatDescFromSYCLDesc(SYCLBufferDesc& desc)
+    {
+        return cv::GMatDesc(desc.depth, desc.chan, desc.size, desc.planar);
+    };
+
+template <typename T, int Dimensions, typename AllocatorT, typename Enable>
+struct GAPI_EXPORTS RMatSYCLBufferAdapter final: public cv::RMat::Adapter
+    {
+        using ReMapDescF = std::function<cv::GMatDesc(const SYCLBufferDesc&)>;
+
+        RMatSYCLBufferAdapter(const sycl::buffer<T, Dimensions, AllocatorT, Enable>& buffer,
+                              const ReMapDescF& bufferDescToGMatDesc):
+          m_buffer(buffer),
+          m_bufferDesc(getSYCLBufferDesc(buffer)),
+          m_bufferDescToGMatDesc(bufferDescToGMatDesc)
```

Comment:
I don't believe you need the ability for the user of the adapter to specify this function. Just call `getGMatDescFromSYCLDesc()` directly from `desc()`?